arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1423,6 +1423,7 @@ Result RestReplicationHandler::processRestoreDataBatch(
     options.ignoreRevs = true;
     options.isRestore = true;
     options.waitForSync = false;
+    options.overwrite = true;
     opRes = trx.insert(collectionName, requestSlice, options);
```

Comment:
is it intended that `trx.insert` calls replace then ?

Reply:
Yes, that was intended.
A few lines further down in RestReplicationHandler.cpp the code will pick up all inserts that failed with error code "unique constraint violated" and then sends them again, this time using `replace`.
Using the `overwrite` flag for the insert will just save this extra step.